### PR TITLE
TST: Fix unsigned pyarrow types in SIGNED_INT_PYARROW_DTYPES

### LINKED ISSUE #48471: BUG: The var SIGNED_INT_PYARROW_DTYPES has incorrect data types
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pyarrow as pa
import pandas as pd
from pandas._testing import SIGNED_INT_PYARROW_DTYPES, UNSIGNED_INT_PYARROW_DTYPES

for pa_dtype in SIGNED_INT_PYARROW_DTYPES:
    assert pa.types.is_signed_integer(pa_dtype)

for pa_dtype in UNSIGNED_INT_PYARROW_DTYPES:
    assert pa.types.is_unsigned_integer(pa_dtype)
```

### Issue Description

There is unsigned data types in SIGNED_INT_PYARROW_DTYPES

### Expected Behavior

No unsigned data types in SIGNED_INT_PYARROW_DTYPES

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 224458ee25d92ccdf289d1ae2741d178df4f323e
python           : 3.9.13.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.6.0
Version          : Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:23 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T6000
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.0rc0
numpy            : 1.23.2
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 65.3.0
pip              : 22.2.2
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.5.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2022.8.2
gcsfs            : None
matplotlib       : 3.5.3
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by gsheni**
- [x] closes #48471 48471
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by datapythonista** [comment]
Thanks @gsheni, is there an issue for this? Can you please provide context about this?

**Comment by mroeschke** [comment]
Thanks for the pull request, but when implementing this I did not specifically include testing `pa.string()` because pyarrow string functionality still fully exists in `StringDtype("pyarrow")` and `ArrowStringArray`. This note will be in the 1.5 docs: https://pandas.pydata.org/docs/dev/reference/arrays.html#pyarrow

What would be helpful is to start an issue to discuss whether there's interest to have `ArrowDtype(pa.string())` replace `StringDtype("pyarrow")` in the future.

**Comment by gsheni** [comment]
@mroeschke sure, issue created here: https://github.com/pandas-dev/pandas/issues/48469

It might be worth reviewing this MR, as the `SIGNED_INT_PYARROW_DTYPES` list has incorrect data types.

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -2212,3 +2212,11 @@ def test_mode(data_for_grouping, dropna, take_idx, exp_idx, request):
     result = ser.mode(dropna=dropna)
     expected = pd.Series(data_for_grouping.take(exp_idx))
     tm.assert_series_equal(result, expected)
+
+
+def test_global_int_lists():
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:2217]: IMO, I don't think we need to test `SIGNED_INT_PYARROW_DTYPES` and `UNSIGNED_INT_PYARROW_DTYPES` have the right items. We don't really test the constants in `pandas/_testing/__init__.py` and it was more of a typo. (Appreciate the gesture though!)

Comment (gsheni) [reply pandas/tests/extension/test_arrow.py]: Removed: https://github.com/pandas-dev/pandas/pull/48457/commits/2f4c7bcf497b89227e109a3dff7fb96bfd903948

**Comment by mroeschke** [comment]
@gsheni mind merging main? Don't remember if these failures were real or unrelated

**Comment by mroeschke** [comment]
`test_add_series_with_extension_array` might need an xfail

File: pandas/tests/extension/test_arrow.py [hunk 2]
```diff
@@ -1619,6 +1619,12 @@ def test_add_series_with_extension_array(self, data, request):
                     reason=f"add_checked not implemented for {pa_dtype}",
                 )
             )
+        elif pa.types.is_signed_integer(pa_dtype):
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:1622]: I think the xfail needs to only be for int8 as the test is failing due to an overflow

Comment (gsheni) [reply pandas/tests/extension/test_arrow.py]: Fixed

**Review by mroeschke — APPROVED** [review]
LGTM. Backporting as `ArrowDtype` support and testing was added in 1.5.

**Comment by mroeschke** [comment]
Thanks @gsheni

**Pull request merged**